server/db.js

```diff
@@ -81,34 +81,51 @@ const fetchRecipeFromPeachworks = (id, callback) => {
   peachworks.proxyGetInventory(id).then(invJson => {
     let inventory = invJson.map(i => {
       return {
-        //TODO: what is quantity vs common_quantity?
         quantity: parseFloat(i.quantity),
         itemId: i.item_id,
-        unitId: i.unit_id
+        unitId: i.unit_id,
+        customUnitId: i.each_unit_id
       };
     });
 
+    const fetchForInv = (idKey, peachworksApi, invKey, jsonKey) => {
+      const ids = inventory.map(i => i[idKey]).filter(i => i);
+
+      return peachworksApi(ids).then(json => {
+        inventory = inventory.map(i => {
+          if (i[idKey]) {
+            const jsonItem = json.find(j => i[idKey] === j.id);
+            //TODO this is pretty gross... any better way to do this?
+            i[invKey] = jsonItem[jsonKey];
```

Comment:
Or is the grossness comment referring to mutating the inventory in place? Or other?